system/lib/wasmfs/wasmfs.cpp

```diff
@@ -369,6 +355,72 @@ __wasi_fd_t __syscall_open(long pathname, long flags, long mode) {
   return FileTable::get().add(openFile);
 }
 
+long __syscall_mkdir(long path, long mode) {
+  auto pathParts = splitPath((char*)path);
+  auto base = pathParts[pathParts.size() - 1];
+
+  // Root directory
+  if (pathParts.empty()) {
+    return -(EEXIST);
+  }
+
+  long err;
+  auto parentDir = getParentDir(pathParts, err);
+
+  if (!parentDir) {
+    // parent node doesn't exist
+    return err;
+  }
+
+  auto lockedParentDir = parentDir->locked();
+
+  auto curr = lockedParentDir.getEntry(base);
+
+  // Check if the request directory already exists.
+  if (curr) {
+    return -(EEXIST);
+  } else {
+    // Create an empty in-memory directory.
+    auto created = std::make_shared<Directory>(mode);
+
+    lockedParentDir.setEntry(base, created);
+    return 0;
+  }
+}
+
+long __syscall_chdir(long path) {
+  auto pathParts = splitPath((char*)path);
+  auto base = pathParts[pathParts.size() - 1];
+
+  // Root directory
+  if (pathParts.empty()) {
```

Comment:
What about `chdir("/")`?

Reply:
(The other code that checks for `"/"` suggests to me that this type of functionality should be shared.)